### PR TITLE
LPD-26755 The URL redirects to the correct language when a category has a translation

### PR DESCRIPTION
**Pull request by boton**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-26755

We are adding a playwright test to check friendly URL redirection to the correct language when a category has a translation

# How am I fixing it?

We are adding a functional test.

# How can you verify that it works?

1. Go to `/modules/test/playwright`
3. Run `npm install` (setup)
4. Run `npm run playwright test -- -g "LPD-26755"` to run the test

# Local test results

## Blogs-web

```sh
npm run playwright test -- --project blogs-web --reporter list
```

image

## Repeat the test

```sh
npm run playwright test -- -g "LPD-26755" --reporter list --repeat-each 10
```

image

> [!IMPORTANT]  
> Unrelated fail in login

# Screenrecord test execution

[blogEntry-LPD-26755-The-URL-redirects-to-the-correct-language-when-a-category-has-a-translation-blogs-web.webm](https://github.com/liferay-content-management/liferay-portal/assets/67901/5fb43663-2784-4567-8862-2189ab4e0bc6)

# UI Screenshot with visible URL
image

---

Added the last commit 099cce86058f00716185e49587c650d5a75582bd to change apiHelpers.headlessAdminTaxonomy to the API ones

For example `postVocabulary` --> `postSiteTaxonomyVocabulary`

image

### CODE REVIEW

**Comment by boton** [comment]
ci:test:playwright

**Comment by boton** [comment]
It's an external error. 
I've added a typo fix but I rollback I prefer to test with more time

**Comment by boton** [comment]
ci:test:playwright

**Comment by boton** [comment]
No failures related to `blogs-web` ✅

File: modules/test/playwright/helpers/HeadlessAdminTaxonomyApiHelper.ts [hunk 1]
```diff
@@ -92,7 +97,7 @@ export class HeadlessAdminTaxonomyApiHelper {
 	 * @param siteId the id of the site in which the tag will be created
 	 */
 
-	async postTag({name, siteId}: postTagProps): Promise<{id: number}> {
+	async postSiteKeyword({name, siteId}: postTagProps): Promise<{id: number}> {
```
Comment (robertoDiaz) [inline modules/test/playwright/helpers/HeadlessAdminTaxonomyApiHelper.ts:100]: Shouldn't we also rename `postTagProps` param to `postSiteKeywordProps`?

File: modules/test/playwright/tests/blogs-web/utils/createCategories.ts [hunk 2]
```diff
@@ -4,6 +4,9 @@
  */
 
 import type {ApiHelpers} from '../../../helpers/ApiHelpers';
+import type {postCategoryProps} from '../../../helpers/HeadlessAdminTaxonomyApiHelper';
```
Comment (robertoDiaz) [inline modules/test/playwright/tests/blogs-web/utils/createCategories.ts:7]: Apply new naming pattern as well to match `postSiteTaxonomyVocabulary `

**Review by robertoDiaz — CHANGES_REQUESTED** [review]
Changes look good to me.

My only concern is related with parameter and attribute names to match the new pattern.

Please change this and forward the PR :)

**Comment by boton** [comment]
ci:test:playwright

**Comment by boton** [comment]
Rebased, nuked the DB and files and re-test and all looks good ✅ 

image

@robertoDiaz I applied your feedback ca39d5c5464b8fb0d6d3db8c872d2374fe84317d 😄  Can you review it?

**Review by robertoDiaz — APPROVED** [review]

**Comment by robertoDiaz** [comment]
ci:forward

**Pull request closed without merging**